packages/web-components/fast-foundation/src/menu/menu.ts

```diff
@@ -139,6 +142,10 @@ export class Menu extends FASTElement {
             if (item === focusIndex) {
                 this.menuItems[item].setAttribute("tabindex", "0");
             }
+            if (hasRole) {
+                this.menuItems[item].classList.remove("without-role");
```

Comment:
Can you just have a single class that you toggle?  Make one state the default and add the class otherwise.

Reply:
I actually think that these visuals may be different in any given design system. I think we need explicit classes for checkbox or radio visuals.

Reply:
These classes control spacing in all menu items based on whether there are any radio/check items in the entire list, so it should be consistent for all menu item types.

Reply:
I don't necessarily agree with this - I could be swayed, but this is confusing to me. What happens if I want to style them differently? How do I accomplish that? I could apply standard spacing to both by chaining classes together in CSS. If I for some reason need them to have different styles, how would I accomplish that? 

Technically, if this just controls spacing for checkbox and radio items, do we need the class selector? It seems like we could control that in CSS using attribute selectors since those attributes will *always* be in the DOM. Perhaps this isn't actually necessary?

Reply:
The purpose of the mechanism is to allow standard menu items to add the space when they share a menu with radio/checkbox items so alignement can be maintained without adding the space when there are not radio/checkbox items in the list. So somehow the menu scans it children and sets some flag that its children can use for styling.  Maybe a class isn't the best way to do this.

Reply:
My concern is that we're adding foundational behaviors to manage a specific visual style that may not be a requirement. If we need a hook to style radio/checkbox implementations, I'd prefer to use those names because it is more semantic and intuitive if I'm trying to target *just* a radio or checkbox, or even both. Alternatively, do we need the hook? Could we simply do something like below in CSS? 

```css
::slotted(fast-menu-item) {
    // add space
}

// can we increase specificity here?
::slotted([role="menuitemcheckbox"]),
::slotted([role="menuitemradio"]) {
    // don't add space
}
```